### ISSUE TITLE
add ability to override "terragrunt-download" directory name

### ISSUE THREAD

**Issue by tamsky** [gruntwork-io/terragrunt#151]
Currently
```
	downloadDir := util.JoinPath(os.TempDir(), "terragrunt-download", encodedWorkingDir, rootPath)
```
https://github.com/gruntwork-io/terragrunt/blob/727b311268f84276cb9d8c68ac0e0ffd1d415a7f/cli/download_source.go#L198

In a multi-user environment this can cause permissions problems.

It would be helpful to be able to declare a per-UID path component instead of the fixed `terragrunt-download` string.

Perhaps an ENV var?

**Comment by brikis98** [comment]
> In a multi-user environment this can cause permissions problems.

Good point.

> It would be helpful to be able to declare a per-UID path component instead of the fixed terragrunt-download string.

A UID is a reasonable idea, but easy to forget. It would be nice to find an option that works automatically. Some ideas:

* Does everyone in the shared environment use different OS users? If so, the username could be inserted in the path to resolve this automatically. 

* Does ever everyone in the shared environment use different AWS credentials? If so, the AWS user id could be inserted in the path to resolve this automatically.

* What about claiming the DynamoDB lock *before* downloading the source? Then, if two users are on the same computer, one of them will get a lock and the other will have to wait.

**Comment by tamsky** [comment]
>Does everyone in the shared environment use different OS users? If so, the username could be inserted in the path to resolve this automatically.

Yes. This.

Different users create directories in /tmp that other users cannot delete or update.

Appending `$USER` would seem to be the most straightforward way to avoid sharing a directory in `/tmp`.

**Comment by brikis98** [comment]
I'm open to a PR for any of those 3 options :)

**Comment by brikis98** [comment]
Fixed in #516.
